AsyncDisplayKit/ASPagerNode.m

```diff
@@ -159,9 +163,13 @@ - (void)setDataSource:(id <ASPagerDataSource>)dataSource
   if (dataSource != _pagerDataSource) {
     _pagerDataSource = dataSource;
     
-    _pagerDataSourceImplementsNodeBlockAtIndex = [_pagerDataSource respondsToSelector:@selector(pagerNode:nodeBlockAtIndex:)];
-    // Data source must implement pagerNode:nodeBlockAtIndex: or pagerNode:nodeAtIndex:
-    ASDisplayNodeAssertTrue(_pagerDataSourceImplementsNodeBlockAtIndex || [_pagerDataSource respondsToSelector:@selector(pagerNode:nodeAtIndex:)]);
+    if (dataSource == nil) {
+      memset(&_pagerDataSourceFlags, 0, sizeof(_pagerDataSourceFlags));
+    } else {
+      _pagerDataSourceFlags.nodeBlockAtIndex = [_pagerDataSource respondsToSelector:@selector(pagerNode:nodeBlockAtIndex:)];
+      // Data source must implement pagerNode:nodeBlockAtIndex: or pagerNode:nodeAtIndex:
+      ASDisplayNodeAssertTrue(_pagerDataSourceFlags.nodeBlockAtIndex || [_pagerDataSource respondsToSelector:@selector(pagerNode:nodeAtIndex:)]);
```

Comment:
@Adlai-Holler what happens if the data source does not implement either of these, in production?  Can we guard against this by setting it to nil or ensuring that nodeAtIndex: isn't called later?